src/app/clusters/time-synchronization-server/TimeSyncManager.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+#include "TimeSyncManager.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandResponseHelper.h>
+#include <app/util/af.h>
+#include <list>
+
+using namespace chip::app::Clusters::TimeSynchronization;
+
+void TimeSyncManager::HandleTimeZoneChanged(TimeZoneList timeZoneList)
+{
+    // do nothing
+    ChipLogProgress(Zcl, "%s", __FUNCTION__);
```

Comment:
I want to leave this for app developers. This is just a placeholder implementation. May be there is a better way to do this?

Reply:
Some of these are already realistically available - ex we have an IPAddress.h that's used to parse IP addresses, and it's cross-platform. That could be implemented in one place with no problems. Ditto for lookups - that's just handling an attribute. 

In my mind, there's kind of a separation between the info that needs to go into the server from the internal impl and externally generated actions that need to be communicated up. 

The places where I think a manufacturer might have some changes from internally rather than a controller:
- time - if they're using NTP or cloud or GNSS or something - probably should go through a similar interface to SetUTCTime (though an internal view with forced update and no response). That way the offset is visible also to the external readers
- time zone - might be changed by something like an internal screen - same thing, want some kind of internal version of the function to be callable here
- dst update - as above, or coming from a time zone change when the node has a database
- defaultNTP - probably fair game for the device to set this itself?

Things that might come from an external source that the manufacturer might want to know are actually fairly similar to the events
- external time zone or DST change 
- time zone or DST expiry
- failures to get time

Things that probably have two-way interactions
- stepping through the time sources list

So I imagine you may want to have callable functions that interact with the server, and also some kind of delegate that gets called for important info. Things like HandleGetDstOffset, HandleDstOffsetLookup, IsNTPAddressValid, IsNTPAddressDomain are things that can be handled in a more general way.

Reply:
The hook for the application to set the UTCTime should actually be a direct set of the offset since NTP will already have accounted for network delay and you don't want to add anything back by piping an old time through a function.